### PR TITLE
Adds more tests, fixes issue with dates

### PR DESCRIPTION
**Pull request by FieryCod**

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -21,3 +21,4 @@ rules:
     no-use-before-define: ["error", { "functions": false }]
     no-unused-vars: ["error", {"argsIgnorePattern":"^next"}]
     comma-dangle: ["error", "never"]
+    no-restricted-syntax: 0
```
Comment (k911) [inline .eslintrc.yml]: we will end with disabling every single eslint option.

Comment (FieryCod) [inline .eslintrc.yml]: When Not To Use It (no-restricted-syntax)
If you don’t want to restrict your code from using any JavaScript features or syntax, you should not use this rule.

Linter was complaining about using generators which are in my opinion fine. I'm not scared of `es6` that is why I've disabled it.

Comment (k911) [inline .eslintrc.yml]: still this rule protects against other things, so you just could change it to:

no-restricted-syntax: ["error", "LabeledStatement", "WithStatement"]

Comment (FieryCod) [inline .eslintrc.yml]: `LabeledStatement` is a little bit hardcore one and I don't think someone uses it these days. But `WithStatement` is sometimes useful. I agree that `LabeledStatement` should output error.

**Pull request merged**